controllers/cron_rollout_helpers.go

```diff
@@ -120,13 +126,29 @@ func createBaseDisruption(owner metav1.Object, disruptionSpec *chaosv1beta1.Disr
 }
 
 // setDisruptionAnnotations updates the annotations of a given Disruption object with those of its owner.
-// Additionally, it sets a scheduled time annotation using the provided scheduledTime.
-func setDisruptionAnnotations(disruption *chaosv1beta1.Disruption, owner metav1.Object, scheduledTime time.Time) {
-	for k, v := range owner.GetAnnotations() {
+// It sets a scheduled time annotation using the provided scheduledTime.
+// It parses the UserInfo annotation if it exists and sets user-related annotations.
+func setDisruptionAnnotations(disruption *chaosv1beta1.Disruption, owner metav1.Object, scheduledTime time.Time, log *zap.SugaredLogger) {
```

Comment:
In the chaos-controller we are not able to unit tests. What I propose is to use encapsulation to update the disruption annotation by using setters:
- `SetAnnotationOwner(object metav1.Object)`
- `SetAnnotationScheduledAt(schedule time.Time)` 
- `SetAnnotationUserMetadata(userinfo string)` 
Those methods are linked to the disruption struct defined in the v1beta1 package. It will be possible to add unit to cover their behavior.

Reply:
I have refactored the code. I adapted the names a little bit, so let me know what you think.